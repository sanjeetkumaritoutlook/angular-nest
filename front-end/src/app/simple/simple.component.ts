import { Component } from '@angular/core';

@Component({
  selector: 'app-simple',
  template: `<h1>{{ title }}</h1>`,
  styleUrl: './simple.component.scss'
})
export class SimpleComponent {
  title = 'Snapshot Testing';
}
