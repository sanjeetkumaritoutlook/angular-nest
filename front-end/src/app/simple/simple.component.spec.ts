import { ComponentFixture, TestBed } from '@angular/core/testing';
//import { render } from '@testing-library/angular';

import { SimpleComponent } from './simple.component';

describe('SimpleComponent', () => {
  let component: SimpleComponent;
  let fixture: ComponentFixture<SimpleComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [SimpleComponent]
    })
    .compileComponents();
    
    fixture = TestBed.createComponent(SimpleComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should be defined', () => {
    expect(component).toBeDefined();
  });
  //snapshot test using nativeElement: option 1
  it('should match snapshot', () => {
    expect(fixture.nativeElement.innerHTML).toMatchSnapshot();
  });

  //snapshot test using render: option 2
  // it('should match snapshot', async () => {
  //   const { container } = await render(SimpleComponent);
  //   expect(container).toMatchSnapshot();
  // });
});
