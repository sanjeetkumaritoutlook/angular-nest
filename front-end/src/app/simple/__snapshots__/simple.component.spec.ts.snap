// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleComponent should match snapshot 1`] = `"<h1>Snapshot Testing</h1>"`;
